flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/table/KafkaDynamicTableFactory.java

```diff
@@ -148,7 +148,15 @@ public String factoryIdentifier() {
     public DynamicTableSource createDynamicTableSource(Context context) {
         final TableFactoryHelper helper = FactoryUtil.createTableFactoryHelper(this, context);
 
-        final ReadableConfig tableOptions = helper.getOptions();
+        final ReadableConfig tableOptions =
+                helper.forwardOptions(
+                                TOPIC,
+                                TOPIC_PATTERN,
+                                SCAN_STARTUP_MODE,
+                                SCAN_STARTUP_SPECIFIC_OFFSETS,
+                                SCAN_TOPIC_PARTITION_DISCOVERY,
+                                SCAN_STARTUP_TIMESTAMP_MILLIS)
+                        .getOptions();
```

Comment:
let's call getOptions() after calling `validate`. maybe we should not offer them as a return type of `forwardOptions`.

Reply:
`forwardOptions` doesn't return anything, it's just a fluent method.